src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -132,6 +132,60 @@ sealed class NumericRange[T](
   // are forgiving: therefore the checks are with the methods.
   private def locationAfterN(n: Int): T = start + (step * fromInt(n))
 
+  private def crossesTheEndAfterN(n: Int): Boolean = {
+    // if we're sure that subtraction in the context of T won't overflow, we use this function
+    // to calculate the length of the range
+    def unsafeRangeLength(r: NumericRange[T]): T = {
+      val diff = num.minus(r.end, r.start)
+      val quotient = num.quot(diff, r.step)
+      val remainder = num.rem(diff, r.step)
+      if (!isInclusive && num.equiv(remainder, num.zero)) quotient else num.plus(quotient, num.one)
+    }
+    val stepIsInTheSameDirectionAsStartToEndVector =
+      (num.gt(end, start) && num.sign(step) == num.one) || (num.lt(end, start) && num.sign(step) == -num.one)
+
+    if (num.equiv(start, end) || n <= 0 || !stepIsInTheSameDirectionAsStartToEndVector) return n >= 1
+
+    val smallerThanInt = num match {
+      case _: Numeric.ShortIsIntegral | _: Numeric.ByteIsIntegral => true
```

Comment:
This won't work for small custom types; this is why `count` does it a different way (actually testing whether `Int.MaxValue` survives a bidirectional trip).  If the algorithm works regardless, maybe just drop this case entirely?

Reply:
It doesn't look like it will work regardless, but maybe defer this comparison to the end?

Reply:
Hmm...Yeah makes sense. I can think of a function which detects whether a `T` fits in `Int`s, until this point we know the length in `T`, so the function can do the trick.